src/Controller/EnMarche/MembershipController.php

```diff
@@ -67,10 +67,44 @@ public function registerAction(Request $request, GeoCoder $geoCoder, Authorizati
      *
      * @Route("/adhesion", name="app_membership_join")
      * @Method("GET|POST")
-     *
-     * @Security("is_granted('ROLE_USER')")
      */
-    public function joinAction(Request $request, EntityManagerInterface $manager): Response
+    public function adhesionAction(Request $request, GeoCoder $geoCoder): Response
+    {
+        if ($this->isGranted('ROLE_USER')) {
+            return $this->joinAdherent($request);
+        }
+
+        $membership = MembershipRequest::createWithCaptcha(
+            $geoCoder->getCountryCodeFromIp($request->getClientIp()),
+            $request->request->get('g-recaptcha-response')
+        );
+
+        $form = $this
+            ->createForm(AdherentRegistrationType::class, $membership)
+            ->add('submit', SubmitType::class, ['label' => 'Adhérer'])
+            ->handleRequest($request)
+        ;
+
+        try {
+            if ($form->isSubmitted() && $form->isValid()) {
+                $membershipRequestHander = $this->get(MembershipRequestHandler::class);
+                $membershipRequestHander->register($membership);
+
+                return $this->redirectToRoute('app_membership_complete');
+            }
+        } catch (ConnectException $e) {
+            $this->addFlash('error_recaptcha', $this->get('translator')->trans('recaptcha.error'));
+        }
+
+        return $this->render('membership/join.html.twig', [
+            'membership' => $membership,
+            'form' => $form->createView(),
+            'countries' => UnitedNationsBundle::getCountries($request->getLocale()),
+            'from_activation' => $request->query->getBoolean('from_activation'),
```

Comment:
It should never happened, this can be set to `false` or add a default value in twig, as you wish